Fix real-time telemetry

akhenry opened the pull request:
<!--- Note: Please open the PR in draft form until you are ready for active review. -->
Closes #144

### Describe your changes:
* Restores the previous realtime-telemetry-provider which is battle-hardened and known to handle WebSocket dropouts reliably. 
* Refactors the realtime-fault-provider to make use of the realtime-telemetryprovider.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Unit tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [ ] Testing instructions included in associated issue?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate unit tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Commit messages meet standards?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

jvigliotta commented on src/providers/realtime-provider.js:
```diff
@@ -86,98 +99,172 @@ export default class RealtimeProvider {
+        this.reconnectTimeout = setTimeout(() => {
+            this.connect();
+            delete this.reconnectTimeout;
+        }, FALLBACK_AND_WAIT_MS[this.currentWaitIndex]);
```
This is cool!

jvigliotta commented on src/providers/realtime-provider.js:
```diff
@@ -86,98 +99,172 @@ export default class RealtimeProvider {
+                console.warn("Error while attempting to send to websocket. Reconnecting...");
+
+                this.requests.push(request);
+                this.reconnect();
```
Do we need to return false here? I see you're returning true above. Is anything relying on this? I didn't see anything, if that's true, you may be able to remove that `return true` above.

jvigliotta commented on src/providers/realtime-provider.js:
```diff
@@ -86,98 +99,172 @@ export default class RealtimeProvider {
+        let wsUrl = `${this.url}`;
+        this.lastSubscriptionId = 1;
+        this.connected = false;
+        this.socket = new WebSocket(wsUrl);
```
I know we're overwriting this here, but do we need to set to undefined or anything first? I know we have listeners attached, I think they get removed on error or close, but just checking.

jvigliotta commented on src/providers/realtime-provider.js:
```diff
@@ -86,98 +99,172 @@ export default class RealtimeProvider {
+    flushQueue() {
+        let shouldReconnect = false;
+        this.requests = this.requests.filter((request) => {
+            try {
```
Cool use of filter here.

jvigliotta reviewed (changes requested):
Lookin nice! Lot's of cool stuff in this. Seems pretty rock solid. Just had a couple suggestions and questions, nothing major.

akhenry commented:
@jvigliotta I believe I have addressed all of these comments, but due to what appears to be a GitHub bug all of my comments are stuck in a `pending` state.
